### ISSUE TITLE
moreh_norm rank-1 W-path crashes on explicit dim=0 reductions

### ISSUE THREAD

**Issue by peter941221** [tenstorrent/tt-metal#46043]
Rank-1 `moreh_norm` crashes on the W-path even when the reduction dimension is explicitly `dim=0`.

This repro is separate from the earlier `dim=None` full-reduction bug class in `moreh_sum`. It fails on a direct 1D reduction.

What I reproduced:

1. Rank-1 input shape `[5]` crashes with:
   - `p=2.0, dim=0, keepdim=True`
   - `p=2.0, dim=0, keepdim=False`
   - `p=0.0, dim=0, keepdim=True`
   - `p=inf, dim=0, keepdim=True`
   - `p=-inf, dim=0, keepdim=True`
2. The pre-patch failure is:
   - `ShapeBase[] index out of range. 1 not in [-4, 1)`

Minimal repro:

```python
import torch
import ttnn

device = ttnn.open_device(device_id=0)
x = torch.tensor([2.0, -1.0, 0.5, 4.0, -3.0], dtype=torch.bfloat16)
tx = ttnn.from_torch(
    x,
    device=device,
    dtype=ttnn.bfloat16,
    layout=ttnn.TILE_LAYOUT,
    pad_value=float("nan"),
)
ttnn.operations.moreh.norm(tx, p=2.0, dim=0, keepdim=True)
ttnn.close_device(device)
```

Code evidence:

1. `moreh_abs_pow` derives `input_rank` from the padded shape, then reads `origin_w` from `logical_shape()[input_rank - 1]`:
   - `ttnn/cpp/ttnn/operations/moreh/moreh_abs_pow/device/moreh_abs_pow_program_factory.cpp:38-49`
2. `moreh_norm` W-path does the same thing:
   - `ttnn/cpp/ttnn/operations/moreh/moreh_norm/device/ord_other/moreh_norm_program_factory_w_other.cpp:31-42`
3. On a logical rank-1 tensor, the padded-rank coordinate can be wider than the logical shape, so that width lookup can index past the logical rank.

Candidate fix:

1. Lift the logical shape to rank 2 before reading `origin_w` in both W-path descriptors.
2. Add focused nightly coverage for rank-1 `dim=0` `moreh_norm` cases.

Validation notes:

1. I reproduced the pre-patch failure in Linux-side emulation.
2. I prepared a narrow source fix in:
   - `ttnn/cpp/ttnn/operations/moreh/moreh_abs_pow/device/moreh_abs_pow_program_factory.cpp`
   - `ttnn/cpp/ttnn/operations/moreh/moreh_norm/device/ord_other/moreh_norm_program_factory_w_other.cpp`
   - `tests/ttnn/nightly/unit_tests/operations/moreh/test_moreh_norm.py`
3. My local emule tree hit unrelated JIT header drift after that point, so I am filing this as a product bug with a narrow source fix rather than as an emulation-environment issue.

**Comment by peter941221** [comment]
I put up a PR for this:
https://github.com/tenstorrent/tt-metal/pull/46044

Patch scope stays narrow:
1. `moreh_abs_pow` now lifts logical rank-1 shapes before reading `origin_w`.
2. The `moreh_norm` W-path descriptor does the same low-rank bridge before its width lookup.
3. The PR adds focused nightly coverage for rank-1 `dim=0` `moreh_norm` cases.

I kept the public diff limited to the source fix and regression coverage. I did not include the local emulation-only helpers I used while reducing the repro in my environment.

**Comment by bbradelTT** [comment]
https://github.com/tenstorrent/tt-metal/pull/46044 is merged. Closing.
